In [4]:
## Following lines such that modules are automatically updated:
%load_ext autoreload
%autoreload 2

import matplotlib.pyplot as plt
import numpy as np
import os, sys
import popoff  # custom package with all code routines
import pop_off_plotting as pop  # module with plotting functions
import pop_off_functions as pof  # module with analysis functions
from Session import SessionLite, Session  # Class object that holds data of 1 recording session
from linear_model import PoolAcrossSessions, LinearModel, MultiSessionModel  # Classes that hold multiple Sessions
import loadpaths 

pathdict = loadpaths.loadpaths()
pathdict

# path_to_vape = os.path.expanduser('~/Documents/code/Vape')
sys.path.append(pathdict['vape_path'])
sys.path.append(os.path.join(pathdict['vape_path'], 'jupyter'))
sys.path.append(os.path.join(pathdict['vape_path'], 'utils'))

# # Folder containing mice folders with individual run pkls
# pkl_path = '/home/jrowland/Documents/code/Vape/run_pkls'
# pkl_path = pathdict['pkl_path']
pkl_path = '/home/tplas/data/mice/runs/'


import utils_funcs as utils
import run_functions as rf
from subsets_analysis import Subsets
import pickle

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [16]:
## Load data
sessions = {}

def only_numerics(seq):
    seq_type= type(seq)
    return seq_type().join(filter(seq_type.isdigit, seq))

def load_files(save_dict, data_dict, folder_path):
    total_ds = 0
    for mouse in data_dict.keys():
        for run_number in data_dict[mouse]:
            try:  # try/except framework to filter datasets that don't work - for whatever reason
                save_dict[total_ds] = Session(mouse, run_number, folder_path, pre_gap_seconds=0,
                                              post_gap_seconds=0, post_seconds=8)  # use_spks=False
                total_ds += 1
                print(f'succesfully loaded mouse {mouse}, run {run_number}')
            except AttributeError:
                pass
#                 print(f'error in mouse {mouse}, run {run_number}')
    return save_dict, total_ds


In [19]:

## Select all possible mouse/run combos:
# pkl_path = '/mnt/qnap_jrowland/run_pkls'
# all_mice = [x for x in os.listdir(pkl_path) if x[-4:] != '.pkl']
# run_dict = {m: list(np.unique([int(only_numerics(x)) for x in os.listdir(pkl_path + f'/{m}')])) for m in all_mice}
# run_dict = {'J064' : [10],
#             'RL048': [23]}
# sessions, total_ds = load_files(save_dict=sessions, data_dict=run_dict, folder_path=pkl_path)

all_mice = [x for x in os.listdir(pkl_path) if x[-4:] != '.pkl']
# all_mice = ['J048', 'RL048']
run_dict = {m: list(np.unique([int(only_numerics(x)) for x in os.listdir(pkl_path + f'/{m}')])) for m in all_mice}
#run_dict = {'J048': [27], 'RL070': [28, 29]}
#run_dict = {'RL070': [29]}

print(all_mice)
print(run_dict)

assert sum([len(m) for m in run_dict.values()]) == 11
sessions, total_ds = load_files(save_dict=sessions, data_dict=run_dict, folder_path=pkl_path)
     
print('The following sessions were loaded:')
for _, ss in sessions.items():
    print(ss, '- frequency:', ss.frequency)

['RL070', 'RL117', 'RL116', 'RL123', 'J064']
{'RL070': [28, 29], 'RL117': [26, 29, 30], 'RL116': [32, 33], 'RL123': [22], 'J064': [10, 11, 14]}
Now loading mouse RL070, run 28
/home/tplas/data/mice/runs/RL070/run28.pkl
microcontroller trial starts occur on average -70.14 ms from galvo trial starts
session RL070_R28: 592 kept, 33 filtered, 625 total
33 / 625 cells filtered
succesfully loaded mouse RL070, run 28
Now loading mouse RL070, run 29
/home/tplas/data/mice/runs/RL070/run29.pkl
microcontroller trial starts occur on average nan ms from galvo trial starts
session RL070_R29: 683 kept, 35 filtered, 718 total
35 / 718 cells filtered
succesfully loaded mouse RL070, run 29
Now loading mouse RL117, run 26
/home/tplas/data/mice/runs/RL117/run26.pkl
microcontroller trial starts occur on average nan ms from galvo trial starts
session RL117_R26: 426 kept, 32 filtered, 458 total
32 / 458 cells filtered
succesfully loaded mouse RL117, run 26
Now loading mouse RL117, run 29
/home/tplas/data/mic

In [18]:
sessions

{}

Overview Rowland et al. cells filtering (max $\Delta F/F$ > 10):
- session J064_R10: 471 kept, 6 excluded, 477 total
- session J064_R11: 788 kept, 14 excluded, 802 total
- session J064_R14: 919 kept, 62 excluded, 981 total
- session RL070_R28: 592 kept, 33 excluded, 625 total
- session RL070_R29: 683 kept, 35 excluded, 718 total
- session RL117_R26: 426 kept, 32 excluded, 458 total
- session RL117_R29: 388 kept, 119 excluded, 507 tot
- session RL117_R30: 529 kept, 204 excluded, 733 total
- session RL116_R32: 300 kept, 45 excluded, 345 total
- session RL116_R33: 329 kept, 65 excluded, 394 total
- session RL123_R22: 494 kept, 58 excluded, 552 total